src/index.ts

```diff
@@ -564,6 +573,63 @@ class threads {
     });
   };
 
+  @muxrpc('async')
+  public recentHashtags = (opts: RecentHashtagsOpts, cb: CB<Array<string>>) => {
+    if (typeof opts.limit !== 'number' || opts.limit <= 0)
+      return cb(new Error('Limit must be number greater than 0'));
+
+    const preserveCase = !!opts.preserveCase;
+
+    // completely normalized hashtag (no leading # and lowercase) -> partially normalized (no leading #)
+    const result = new Map<string, string>();
+
+    return pull(
+      this.ssb.db.query(
+        where(and(isPublic(), hasSomeHashtag())),
+        descending(),
+        batch(BATCH_SIZE),
+        toPullStream(),
+      ),
+      this.removeMessagesFromBlocked,
+      pull.through((msg: Msg<any>) => {
+        const { channel, mentions } = msg.value.content;
+
+        if (channel) {
+          const withoutHashtag = withoutLeadingHashtag(channel);
+          const lowercaseWithoutHashtag = withoutHashtag.toLocaleLowerCase();
+
+          // Since the messages received are descending,
+          // we don't want to update the value for associated key if it already exists
+          // because we want the keep the most recent variation of the hashtag (accounts for casing)
+          if (!result.has(lowercaseWithoutHashtag)) {
+            result.set(lowercaseWithoutHashtag, withoutHashtag);
+          }
+        } else if (Array.isArray(mentions)) {
+          // Assumes `mentions` goes from oldest to most recent
+          // so we want to process them in reverse order
+          for (const mention of mentions.reverse()) {
```

Comment:
i _think_ it makes sense to reverse this, assuming that most clients will publish the `mentions` field as `[first, ..., last]`

Reply:
I think we shouldn't, because these `mentions` here are for a single message, which means they were published at the same time and doesn't make sense to order them. The `reverse()` though incurs a little bit of a performance hit, so it doesn't come for free.

Could you remove `.reverse()`?